packages/realm/src/errors.ts

```diff
@@ -138,3 +140,45 @@ export class ClientResetError extends SyncError {
     };
   }
 }
+
+/**
+ * An error class that indicates that one or more object changes have been reverted by the server.
+ * This can happen when the client creates/updates objects that do not match any subscription, or performs writes on
+ * an object it didn't have permission to access.
+ */
+export class CompensatingWriteError extends SyncError {
+  /**
+   * The array of compensating writes performed by the server.
```

Comment:
```suggestion
   * The array of compensating write failures performed by the server.
```

Reply:
As proven by the tests, only the failed writes are included in the error.

Reply:
ah, I have misunderstood what CompensatingWrite actually means.  The compensating write, is the server telling the client to remove those entries that it didn't have the rights to perform.  The comment is correct then.